ckanclient/__init__.py

```diff
@@ -167,12 +170,12 @@ def push_blob(self, resource):  # TODO async version with on_progress attr
         result = {
             'oid': obj['oid'],
             'size': obj['size'],
-            'name': resource['descriptor']['name'],
+            'name': resource['name'],
             'success': True,
             'fileExists': True,
         }
 
-        if not obj["actions"]:  # File is already in storage
+        if not obj.get("actions"):  # File is already in storage
             return result
 
         result["file_exists"] = False
```

Comment:
@cuducos there is no result returned in this case (new file created) atm, so you always will get `None`